postgres/tests/test_pg_integration.py

```diff
@@ -483,6 +490,13 @@ def test_statement_samples_collect(
             # we expect to get a duration because the connections are in "idle" state
             assert event['duration']
 
+        # validate the events (if applicable) to ensure we've provided an explanation for not providing an exec plan
+        for event in matching:
+            if event['db']['plan']['definition'] is None:
+                assert event['db']['plan']['collection_error'] != {}
```

Comment:
Where possible we should validate that we're producing exactly the expected `collection_error`